gui_scripts/gui_helpers/general_helpers.py

```diff
@@ -202,3 +202,178 @@ def stop(self):
             self.paused = False
             self.output_hints_signal.emit('Stopping simulation from thread')
         self.pause_condition.wakeOne()
+
+
+class DirectoryTreeView(QtWidgets.QTreeView):
+    """
+    Sets up a new directory tree view
+    """
+    item_double_clicked_sig = QtCore.pyqtSignal(QtCore.QModelIndex)
+
+    def __init__(self, file_model, parent=None):
+        super().__init__(parent)
+        self.setSelectionBehavior(QtWidgets.QTreeView.SelectRows)
+        self.setSelectionMode(QtWidgets.QTreeView.SingleSelection)
+
+        self.model = file_model
+        self.setModel(self.model)
+
+        self.copied_path = None
+        self.is_directory = False
+        self.is_cut_operation = False
+
+    def copy_item(self, source_index, cut=False):
+        """
+        Implements the copy/cut operations in the directory tree view
+
+        :param source_index:    the QModelIndex of the item to be copied or cut
+        :param cut:             boolean to choose between copying and cutting
+        """
+        # Store the source path and determine if it's a directory
+        self.copied_path = self.model.filePath(source_index)
+        self.is_directory = QtCore.QFileInfo(self.copied_path).isDir()
+        self.is_cut_operation = cut
+
+    def copy_directory(self, source_dir, destination_dir):
+        """
+        Copies a directory recursively
+
+        :param source_dir:
+        :param destination_dir:
+        """
+        source = QtCore.QDir(source_dir)
+        destination = QtCore.QDir(destination_dir)
+
+        destination_path = destination.filePath(
+            QtCore.QFileInfo(source_dir).fileName())
+        if not destination.exists(destination_path):
+            destination.mkpath(destination_path)
+
+        for file_name in source.entryList(QtCore.QDir.Files):
```

Comment:
Source looks like an object, I think it also says in our coding guidelines we must name variables if they involve data structures. For example a dictionary, list, object, etc. Usually, we name this something like “source_obj”.